subprojects/docs/src/docs/userguide/commandLineInterface.adoc

```diff
@@ -0,0 +1,520 @@
+
+`-Dorg.gradle.logging.level=(quiet warn lifecycle info debug)`::
+Set logging level via Gradle properties.
+
+`-q`, `--quiet`::
+Log errors only.
+
+`-w`, `--warn`::
+Set log level to warn.
+
+`-i`, `--info`::
+Set log level to info.
+
+`-d`, `--debug`::
+Log in debug mode (includes normal stacktrace).
+
+Lifecycle is the default log level.
+
+==== Customizing Log Format
+You can control the use of rich output (colors and font variants) by specifying the "console" mode in the following ways:
+
+`-Dorg.gradle.console=(auto plain rich verbose)`::
+Specify console mode via Gradle properties. Different modes described immediately below.
+
+`--console=(auto plain rich verbose)`::
+Specifies which type of console output to generate.
++
+Set to `plain` to generate plain text only. This option disables all color and other rich output in the console output.
++
+Set to `auto` (the default) to enable color and other rich output in the console output when the build process is attached to a console, or to generate plain text only when not attached to a console.
++
+Set to `rich` to enable color and other rich output in the console output, regardless of whether the build process is not attached to a console. When not attached to a console, the build output will use ANSI control characters to generate the rich output.
++
+Set to `verbose` to enable color and other rich output like the `rich`, but output task names and outcomes at the lifecycle log level, as is done by default in Gradle 3.5 and earlier.
+
+==== Rich Console
+
+Gradle's rich console displays extra information while builds are running.
+
+image::img/console-animation.gif[alt="Gradle Rich Console",data-alt="foo.gif",class="js-gif-playback-control gif-control gif-control__paused"]
+
+Features:
+
+ * Logs above grouped by task that generated them
+ * Progress bar and timer visually describe overall status
+ * Parallel work-in-progress lines below describe what is happening now
+
+=== Execution Options
+The following options affect how builds are executed, by changing what is built or how dependencies are resolved.
+
+`-a`, `--no-rebuild` (deprecated)::
+Do not rebuild project dependencies.
+
+`--include-build`::
+Run the build as a composite, including the specified build. See <<composite_builds, Composite Builds>>.
+
+`--offline`::
+Specifies that the build should operate without accessing network resources. Learn more about <<cache_command_line_options,options to override dependency caching>>.
+
+`--refresh-dependencies`::
+Refresh the state of dependencies. Learn more about how to use this in the dependency management docs.<<cache_command_line_options, dependency management docs>>.
+
+`--dry-run`::
+Run Gradle with all task actions disabled. Use this to show which task would have executed.
+
+[[sec:command_line_bootstrapping_projects]]
+=== Bootstrapping New Projects
```

Comment:
:o: I am not sure how I feel about having this chapter under "Command-Line Interface" at all. I would expect to find this information here.

Reply:
Did you mean "I would _not_ expect to find this information here."? 

One alternative I think we could consider is documenting options added by core plugins like init and wrapper under a different header. One might say "this belongs in the init chapter" and it does of course, but I argue that `init` and `wrapper` are so core to Gradle that they belong in the CLI reference.

Reply:
My comment was about some of the content in this chapter that somewhat didn't feel CLIish but rather a core functionality of the Gradle e.g. `init`.

Reply:
I think this stuff does focus on the CLI of the init and wrapper plugins, and not so much on the details, which is linked here. My argument is that duplication is okay here because those 2 things are _so core_ to Gradle.